replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/data/RepSyncImpl.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl.data;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.codice.ditto.replication.api.Direction;
+import org.codice.ditto.replication.api.ReplicationType;
+import org.codice.ditto.replication.api.ReplicatorConfig;
+
+/**
+ * RepSyncImpl is an implementation of ReplicatorConfig and has methods that allow it to easily be
+ * converted to, or from, a map.
+ */
+public class RepSyncImpl extends AbstractPersistable implements ReplicatorConfig {
+
+  // public so that the persistent store can access it using reflection
+  public static String PERSISTENCE_TYPE = "replication_repsync";

```

Comment:
I think I might just hard code getType() to return this string and the others like it. That way, in the ReplicationPersistentStore, we can just create an instance of this class and get the type from it without using reflection.

Reply:
Keep in mind that you would be creating an object for no other reasons than to get its type. Using reflection is not an evil thing to do; it is a valid and acceptable practice in many APIs today.

Reply:
My only concern was that when we use reflection our tools can't help us, Intellij can't remind me that I need to add a static variable with this name if I create another persistable, and I'd like to leverage our tools as much as possible. I'm going to leave this alone however because it's not a big deal.